climada/entity/exposures/crop_production.py

```diff
@@ -481,180 +490,201 @@ def init_full_exposure_set(input_dir=INPUT_DIR, filename=None, hist_mean_dir=HIS
-                                      hist_mean=hist_mean_dir, bbox=bbox,
-                                      yearrange=yearrange, crop=((item[2]).split('-'))[0],
-                                      irr=((item[2]).split('-'))[1], unit=unit)
-        filename_saveto = ('cropyield_isimip_' + ((item[2]).split('-'))[0] + '-'
-                           + (((item[2]).split('-'))[1]).split('.')[0] + '_'
-                           + str(yearrange[0]) + '-' + str(yearrange[1]) + '.hdf5')
-        filename_list.append(filename_saveto)
-        output_list.append(cropyield)
-        cropyield.write_hdf5(os.path.join(output_dir, 'Exposure', filename_saveto))
-
-    if returns == 'filename_list':
+    for file in filenames:
+        histmean_prop = file.split('_')
+        crop_production = CropProduction()
+        crop_production.set_from_single_run(input_dir=input_dir, filename=filename,
+                                            hist_mean=hist_mean_dir, bbox=bbox,
+                                            yearrange=yearrange,
+                                            crop=((histmean_prop[2]).split('-'))[0],
+                                            irr=((histmean_prop[2]).split('-'))[1], unit=unit)
+        filename_list.append(('crop_production_' + ((histmean_prop[2]).split('-'))[0] + '-'
+                              + (((histmean_prop[2]).split('-'))[1]).split('.')[0] + '_'
+                              + str(yearrange[0]) + '-' + str(yearrange[1]) + '.hdf5'))
+        output_list.append(crop_production)
+        crop_production.write_hdf5(os.path.join(output_dir, 'Exposure', filename_list[-1]))
+
+    if not return_data:
         return filename_list
     return filename_list, output_list
 
-def normalize_with_fao_cropyield(exp_firr, exp_noirr, input_dir=INPUT_DIR,
-                                 yearrange=np.array([2008, 2018]),
-                                 unit='t', returns='all'):
-    """Normalize the given exposures countrywise with the mean cropyield production quantity
+def normalize_with_fao_cp(exp_firr, exp_noirr, input_dir=INPUT_DIR,
+                          yearrange=np.array([2008, 2018]), unit='t', return_data=True):
+    """Normalize the given exposures countrywise with the mean crop production quantity
     documented by the FAO.
 
-        Parameters:
-        exp_firr (cropyield_isimip): exposure under full irrigation
-        exp_noirr (cropyield_isimip): exposure under no irrigation
+    Parameters:
+        exp_firr (crop_production): exposure under full irrigation
+        exp_noirr (crop_production): exposure under no irrigation
+
+    Optional Parameters:
         input_dir (str): directory containing exposure input data
-        yearrange (array): the mean yield in this year range is used to normalize the exposure
-            data (default 2008-2018)
+        yearrange (array): the mean crop production in this year range is used to normalize
+            the exposure data (default 2008-2018)
```

Comment:
Is the default yearrange chosen by you or related to the datasource?

Reply:
It's chosen by us - should we specify something more about that in the docstring?